python/tank/folder/folder_types.py

```diff
@@ -1679,8 +1694,40 @@ def __init__(self,
                         entity_filter, 
                         create_with_parent=True)
                 
-        
-        
+    def _get_additional_sg_fields(self):
+        """
+        Returns additional shotgun fields to be retrieved.
+
+        Subclassed for tasks so that the step data is always
+        retrieved at the same time as the task data.
+
+        :returns: List of shotgun fields to retrieve in addition to those
+                  specified in the configuration files.
+        """
+        return ["step"]
+
+    def _register_secondary_entities(self, io_receiver, path, entity):
+        """
+        Looks in the entity dict for any linked entities and register these.
```

Comment:
Should this line be updated? Something more specific to `step` instead of `linked entities`?